polyfill/lib/ecmascript.mjs

```diff
@@ -920,15 +920,9 @@ export const ES = ObjectAssign(ObjectAssign({}, ES2019), {
     if (!Number.isFinite(num) || Math.abs(num) !== num) throw new RangeError(`invalid positive integer: ${num}`);
     return num;
   },
-  SystemUTCEpochNanoSeconds: (() => {
-    let ns = Date.now() % 1e6;
-    return () => {
-      const ms = Date.now();
-      const result = bigInt(ms).multiply(1e6).plus(ns);
-      ns = ms % 1e6;
-      return result;
-    };
-  })(),
+  SystemUTCEpochNanoSeconds: () => {
+    return bigInt(Date.now()).multiply(1e6);
```

Comment:
In theory I'd say 0 is better than bogus nanoseconds, but I wonder if it was done this way intentionally so that two immediately consecutive calls would be less likely to return the same value if done within the same millisecond? Definitely when I do `a = Date.now() % 1e6; b = Date.now() % 1e6; console.log(a, b)` in the Node REPL it most often shows the same result.

Reply:
@pipobscure thoughts?

Reply:
That's why I originally added *bogus* nanoseconds that are based on the previous iterations milliseconds. That way there is a guarantee that the full nanoseconds are always going to be increasing at least. The issue being that using *0* makes it easy to write tests that never deal with below ms accuracy. This instead will pretty much ensure that the microsecond and nanosecond fields are likely to actually have a value.

Reply:
Maybe keep the old code and add a comment explaining why this was done?

Although also note that `performance.now()` rounds to 1 ms anyway in order to prevent Spectre/Meltdown attacks.